crates/sui-framework/sources/governance/sui_system.move

```diff
@@ -530,6 +540,21 @@ module sui::sui_system {
             == old(balance::supply_value(self.sui_supply)) + old(stake_subsidy::current_epoch_subsidy_amount(self.stake_subsidy));
     }
 
+    /// An extremely simple version of advance_epoch.
+    /// This is only called when the call to advance_epoch failed due to a bug, and we want to be able to keep the system
+    /// running and continue making epoch changes.
+    public entry fun advance_epoch_safe_mode(
+        self: &mut SuiSystemState,
+        new_epoch: u64,
+        ctx: &mut TxContext,
+    ) {
+        // Validator will make a special system call with sender set as 0x0.
+        assert!(tx_context::sender(ctx) == @0x0, 0);
+
```

Comment:
Do we want to sanity check that `new_epoch == self.epoch + 1` or not event that?

Reply:
I don't see a point asserting it here.